integration/kubernetes/helm-chart/alluxio/templates/job/format-journal-job.yaml

```diff
@@ -17,89 +18,82 @@
 {{- $isSingleUfsLocal := and $isUfsLocal $isSingleMaster }}
 {{- $needJournalVolume := or $isEmbedded $isUfsLocal }}
 {{- $release := .Release }}
+{{- $name := include "alluxio.name" . }}
+{{- $fullName := include "alluxio.fullname" . }}
+{{- $chart := include "alluxio.chart" . }}
+{{- range $i := until $masterCount }}
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: {{ template "alluxio.fullname" . }}-format-master
+  name: {{ $fullName }}-format-master-{{ $i }}
   labels:
-    name: {{ template "alluxio.fullname" . }}-format-master
-    app: {{ template "alluxio.name" . }}
-    chart: {{ template "alluxio.chart" . }}
+    name: {{ $fullName }}-format-master-{{ $i }}
+    app: {{ $name }}
+    chart: {{ $chart }}
     release: {{ $release.Name }}
     heritage: {{ $release.Service }}
     role: alluxio-master
 spec:
-  activeDeadlineSeconds: {{ .Values.journal.format.job.activeDeadlineSeconds }}
-  ttlSecondsAfterFinished: {{ .Values.journal.format.job.ttlSecondsAfterFinished }}
+  activeDeadlineSeconds: {{ $root.Values.journal.format.job.activeDeadlineSeconds }}
+  ttlSecondsAfterFinished: {{ $root.Values.journal.format.job.ttlSecondsAfterFinished }}
   template:
     spec:
       containers:
         - name: alluxio-master
-          image: {{ .Values.image }}:{{ .Values.imageTag }}
-          imagePullPolicy: {{ .Values.imagePullPolicy }}
-          {{- if .Values.master.resources  }}
-          resources:
-            {{- if .Values.master.resources.limits }}
-            limits:
-              cpu: {{ .Values.master.resources.limits.cpu }}
-              memory: {{ .Values.master.resources.limits.memory }}
-            {{- end }}
-            {{- if .Values.master.resources.requests }}
-            requests:
-              cpu: {{ .Values.master.resources.requests.cpu }}
-              memory: {{ .Values.master.resources.requests.memory }}
-            {{- end }}
+          image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
+          imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+          securityContext:
+            runAsUser: {{ $root.Values.user }}
+            runAsGroup: {{ $root.Values.group }}
+          {{- if $root.Values.journal.format.resources }}
+{{ include "alluxio.journal.format.resources" $root | indent 10 }}
           {{- end }}
           command: ["/opt/alluxio/bin/alluxio"]
-          args: ["formatMasters"]
+          args: ["formatJournal"]
           envFrom:
             - configMapRef:
-                name: {{ template "alluxio.fullname" . }}-config
+                name: {{ $fullName }}-config
+          {{- if $isHaEmbedded }}
+          env:
+            - name: ALLUXIO_MASTER_HOSTNAME
+              value: {{ printf "%v-master-%v" $fullName $i }}
+          {{- end }}
```

Comment:
This is required in HA, otherwise the following error will be thrown
```
2020-02-20 15:50:19,510 ERROR Format - Failed to format
java.lang.IllegalStateException: The cluster addresses ([alluxio-master-0:19200, alluxio-master-1:19200, alluxio-master-2:19200]) must contain the local master address (alluxio-format-master-0-2pm8w:19200)
	at com.google.common.base.Preconditions.checkState(Preconditions.java:823)
	at alluxio.master.journal.raft.RaftJournalConfiguration.validate(RaftJournalConfiguration.java:77)
```